### ISSUE TITLE
The unit tests require a large amount of memory

### ISSUE THREAD

**Issue by bowlofeggs** [fedora-infra/bodhi#981]
The unit tests use quite a lot of memory. The current `Vagrantfile.example` sets the guest's memory to 4 GB, but on my dev box the unit tests often fail with a few of them raising `OSError: [Errno 12] Cannot allocate memory`. I've raised my guest's memory to 8 GB. It would be good to figure out what is using all this memory so we can get the required RAM down to a more reasonable level for development.

I have some suspicions that this is due to the use of the in-memory sqlite connection. There may be ways to ask sqlite to clear out data of past tests so that it doesn't continuously grow.

**Comment by jeremycline** [comment]
So I spent a good bit of time investigating this today and reading sqlalchemy docs. I found a few suspicious things which I'll outline here:
- [Repeatedly creating the TestApp seems to leak memory in some way, but I didn't investigate too much. It might be better to create this once for the test suite.](https://github.com/fedora-infra/bodhi/blob/58fc9c55220f165e44400912715647c784e30e95/bodhi/tests/server/functional/base.py#L79)
- [Some per-test setup should definitely not be happening per-test, but moving it breaks tests](https://github.com/fedora-infra/bodhi/blob/58fc9c55220f165e44400912715647c784e30e95/bodhi/tests/server/base.py#L38). 

I think the best way to address this ticket is to start refactoring the tests to use a single superclass and carefully set that up to ensure it doesn't leak memory and always cleans the database out. Right now, there are some tests that aren't using the base classes, and there are quite a few base classes floating around.
